src/BuildScriptGenerator/SdkStorageVersionProviderBase.cs

```diff
@@ -47,46 +44,29 @@ protected PlatformVersionInfo GetAvailableVersionsFromStorage(string platformNam
 
-            foreach (var blobElement in xdoc.XPathSelectElements($"//Blobs/Blob"))
-            {
-                var childElements = blobElement.Elements();
-                if (this.commonOptions.DebianFlavor == OsTypes.DebianStretch)
-                {
-                    var versionElement = childElements
-                        .Where(e => string.Equals("Metadata", e.Name.LocalName, StringComparison.OrdinalIgnoreCase))
-                        .FirstOrDefault()?.Elements()
-                        .Where(e => string.Equals(SdkStorageConstants.LegacySdkVersionMetadataName, e.Name.LocalName, StringComparison.OrdinalIgnoreCase))
-                        .FirstOrDefault();
-
-                    if (versionElement != null)
-                    {
-                        supportedVersions.Add(versionElement.Value);
-                    }
+            var isStretch = string.Equals(this.commonOptions.DebianFlavor, OsTypes.DebianStretch, StringComparison.OrdinalIgnoreCase);
 
-                    continue;
-                }
+            var sdkVersionMetadataName = isStretch
+                ? SdkStorageConstants.LegacySdkVersionMetadataName
+                : SdkStorageConstants.SdkVersionMetadataName;
 
-                // TODO: PR2 Remove this logic, as we will be able to depend on bullseye images existing in the sdk storage
-                var patternText = this.commonOptions.DebianFlavor == OsTypes.DebianBullseye
-                    ? $"{platformName}-{OsTypes.DebianBuster}-(?<version>.*?).tar.gz"
-                    : $"{platformName}-{this.commonOptions.DebianFlavor}-(?<version>.*?).tar.gz";
-
-                // try to parse the version from the file name, as we currently don't supply version metadata to non-stretch sdks
-                // TODO: PR2 Add logic to use SdkStorageConstants.SdkVersionMetadataName and SdkStorageConstants.OsTypeMetadataName
-                // to determine the supported sdks
-                var fileName = childElements
-                        .Where(e => string.Equals("Name", e.Name.LocalName, StringComparison.OrdinalIgnoreCase))
-                        .FirstOrDefault();
-
-                if (fileName != null)
+            foreach (var metadataElement in xdoc.XPathSelectElements($"//Blobs/Blob/Metadata"))
+            {
+                var childElements = metadataElement.Elements();
+                var versionElement = childElements
+                    .Where(e => string.Equals(sdkVersionMetadataName, e.Name.LocalName, StringComparison.OrdinalIgnoreCase))
+                    .FirstOrDefault();
+
+                var osTypeElement = childElements
+                    .Where(e => string.Equals(SdkStorageConstants.OsTypeMetadataName, e.Name.LocalName, StringComparison.OrdinalIgnoreCase))
+                    .FirstOrDefault();
+
+                // if the os type is stretch and we find a blob with the correct version metadata, we add as a supported version
```

Comment:
nit: I'd rephrase this comment so that it prioritizes the fact that the version metadata needs to match since the result of `versionElement != null` is `&&`'d against the rest of the conditional statement, so it must always be `true` if we're to add this as a supported version.